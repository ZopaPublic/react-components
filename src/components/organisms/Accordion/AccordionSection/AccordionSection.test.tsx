import axe from '../../../../../axe-helper';
import React from 'react';
import { render } from '@testing-library/react';
import { Accordion } from '..';

describe('<AccordionSection />', () => {
  it('renders the component with no a11y violations', async () => {
    const { container, getByTestId } = render(
      <Accordion aria-label="test-accordion">
        <Accordion.Header id="one" index={0} textSize="small">
          Header
        </Accordion.Header>
        <Accordion.Section data-automation="section" id="one" index={0}>
          Content
        </Accordion.Section>
      </Accordion>,
    );
    expect(getByTestId('section')).toMatchSnapshot();
    const results = await axe(container.innerHTML);
    expect(results).toHaveNoViolations();
  });
});
