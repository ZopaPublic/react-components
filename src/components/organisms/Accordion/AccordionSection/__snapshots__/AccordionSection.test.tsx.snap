// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AccordionSection /> renders the component with no a11y violations 1`] = `
.c0 {
  visibility: hidden;
  -webkit-transition: visibility 200ms;
  transition: visibility 200ms;
}

<div
  aria-hidden="true"
  aria-labelledby="one"
  data-automation="section"
  id="one-section"
  role="region"
  style="overflow: hidden; transition: height 200ms linear; height: 0px;"
>
  <div
    class="c0"
    tabindex="-1"
  >
    Content
  </div>
</div>
`;
